cg/store/crud/read.py

```diff
@@ -1729,19 +1729,38 @@ def get_all_pools_to_deliver(self) -> list[Pool]:
         return records.all()
 
     def get_orders(self, orders_request: OrdersRequest) -> tuple[list[Order], int]:
-        order_query: Query = apply_order_filters(
+        filtered_orders: Query = self.filter_orders(orders_request)
+        total_count: int = filtered_orders.count()
+        sorted_orders: Query = self.sort_orders(
+            orders=filtered_orders, orders_request=orders_request
+        )
+        orders: list[Order] = self.paginate_orders(
+            orders=sorted_orders, orders_request=orders_request
+        )
+        return orders, total_count
+
+    def filter_orders(self, orders_request: OrdersRequest) -> Query:
+        return apply_order_filters(
             orders=self._get_query(Order),
             filters=[OrderFilter.BY_WORKFLOW],
             workflow=orders_request.workflow,
         )
-        total_count: int = order_query.count()
-        orders: list[Order] = apply_order_filters(
-            orders=order_query,
+
+    def sort_orders(self, orders: Query, orders_request: OrdersRequest) -> Query:
+        return apply_order_filters(
+            orders=orders,
+            filters=[OrderFilter.SORT],
+            sort_field=orders_request.sort_field,
+            sort_order=orders_request.sort_order,
+        )
+
+    def paginate_orders(self, orders: Query, orders_request: OrdersRequest) -> list[Order]:
+        return apply_order_filters(
+            orders=orders,
             filters=[OrderFilter.PAGINATE],
             page=orders_request.page,
             page_size=orders_request.page_size,
         ).all()
```

Comment:
1. It is less readable to have a function accepting 10 optional parameters when they can be grouped naturally in an object.
2. When we filter by many different optional parameters, this leads to very poorly named functions.
3. Will consider moving the functions into the filters module 👍 
4. Yes, that is the point - they are chained.

I don't agree that all the functions should be merged into one. That is a violation of the single responsibility principle and it makes the code less readable.

Reply:
Removed the `filter_orders` and merged the sorting/pagination filters into one, it is cleaner!

Reply:
1. These functions build queries using these paramters. If you abstract that away others cannot immediatly see what the query is doing.
2. Not knowing what the query is doing by its parameters and a generic name will make it even more difficult. Also `get_samples`-> give all samples, `get_cases` -> gives all cases. In this case `get_orders` gets all orders for a workflow sorted and paginated with another `int` in the return.
3. All other CRUD functions have multiple filters in the apply filter functions, since they are building queries. We can simply do:
```python
filter_functions= [OrderFilter.BY_WORKFLOW, OrderFilter.SORT_BY_FIELD, OrderFilter.Paginate]
```
4. The chaining happens in the `filter_functions` list as described in point 3. At least this is how we designed it

Reply:
That won't work, the total count is needed after the filtering

Reply:
Unresolving since the discussion seems ongoing. 

1. Is this necessarily a problem? We could have support for filtering on many different parameters, at which point I would find it clearer to see the allowed parameters in their own object either way.
2. What would be the suggested design for the "get_orders" method? My suggestion would be "get_filtered_orders" to avoid clashing with the convention with "get_cases" for example. At this point "get_orders_by_workflow" would be even unclearer, given that more happens than just a filtering on workflow. I do agree that the return type is a bit weird. Maybe the total can just be calculated as the length of the list afterwards although we would need to think of how to do that with pagination.
3. This will be about 3 and 4 since I can't really find a clear distinction. There is support for sending all the filters in the apply_order_filters method. I have not really seen any guidelines about having to use them in one go, rather than split them up, though. In this case, I think the total can not be calculated when passing all filters, since the pagination truncates the returned entries. Also I have not really seen a method here which returns queries. Is it the filter_orders and sort_orders that we speak of? Are they not essentially filter functions since they wrap an apply_filter method which usually returns queries?

Reply:
Would this be an acceptable compromise to you? This is mainly so that in future we do not get confused when we need to make more queries for orders.

1. better, but not too long name definition: I would now know something special is happened when using `get_orders_by_order_request`.
2. We can pass the `OrderRequest`
3. We can remove `filter_orders` -> I think this would just cause confusion in future. 
4. We keep the needed functionality.
```Python

def get_orders_by_order_request(self, orders_request: OrdersRequest) -> tuple[list[Order], int]:
    orders: Query = apply_order_filters(
        orders=self._get_query(Order),
        filters=[OrderFilter.BY_WORKFLOW],
        workflow=orders_request.workflow,
    )
    total_count: int = orders.count()
    orders: list[Order] = self.sort_and_paginate_orders(
        orders=orders, orders_request=orders_request
    )
    return orders, total_count


def get_sorted_and_paginated_orders(
    self, orders: Query, orders_request: OrdersRequest
) -> list[Order]:
    return apply_order_filters(
        orders=orders,
        filters=[OrderFilter.SORT, OrderFilter.PAGINATE],
        sort_field=orders_request.sort_field,
        sort_order=orders_request.sort_order,
        page=orders_request.page,
        page_size=orders_request.page_size,
    ).all()

```

Reply:
No, I don't like that style of naming. Why would I repeat the name of a parameter in the function name? It is not DRY. 

This review is getting out of hand. If you think a PR adds value, the default is to approve. Always. Unless there is a major issue.

Reply:
> Unresolving since the discussion seems ongoing.
> 
> 1. Is this necessarily a problem? We could have support for filtering on many different parameters, at which point I would find it clearer to see the allowed parameters in their own object either way.

Yes I see both your point. The main thing is that we know what the query is constructed with and that it does return something less or modified than just all orders. I sort of agree that it can be a model with many parameters also. Perhaps having a model that is more descriptive than "OrderRequest" would be good. 
Otherwise this will lead to a lot of cognitive load in understanding what this query will return me when needing to work with it. Perhaps a concise and clear docstring would also help then.

Reply:
> No, I don't like that style of naming. Why would I repeat the name of a parameter in the function name? It is not DRY.

Because in this instance you are building a query to retrieve data. Not being explicit here can mislead others that want to re-use code. If you don't want to re-use the variable name then a descriptive name of what this query is for could be an option